tests/pos/i7383.scala

```diff
@@ -0,0 +1,12 @@
+trait ZSink {
+
+  type State
+
+  def foo: ZSink =
+    class Anon extends ZSink
```

Comment:
I'd add back the braces here to make sure that file doesn't stop working or change meaning if the syntax changes.